backstage/src/executor/activity.ts

```diff
@@ -46,31 +45,8 @@ export function performActivity(gameContext: GameContext, activity: Ident): QRes
    )
 
    if (activityContent.output) {
-      const { attributes, skillPoints, mentalHealth, satisfactory, money } = activityContent.output
-      const propertySource = `@activity:${activityContent.category}`
-      if (attributes) {
-         for (const attrName in attributes) {
-            const attrValue = attributes[<keyof PlayerAttributesUpdate>attrName]
-            if (attrValue) {
-               updatePlayerProperty(gameContext, `attributes.${attrName}`, 'add', attrValue, propertySource)
-            }
-         }
-      }
-
-      if (skillPoints) {
-         updatePlayerProperty(gameContext, 'skillPoints', 'add', skillPoints, propertySource)
-      }
-
-      if (mentalHealth) {
-         updatePlayerProperty(gameContext, 'mentalHealth', 'sub', mentalHealth, propertySource)
-      }
-
-      if (satisfactory) {
-         updatePlayerProperty(gameContext, 'satisfactory', 'add', satisfactory, propertySource)
-      }
-
-      if (money) {
-         updatePlayerProperty(gameContext, 'money', 'add', money, propertySource)
+      for (const propertyId in activityContent.output) {
+         updateProperty(gameContext, propertyId, 'add', activityContent.output[propertyId], '@activity')
```

Comment:
Either automatically determine `add` or `sub` here, or in the implementation of `updateProperty`. Otherwise modifier system will be messed up.